frontend/apps/cloudstorage/ftp.lua

```diff
@@ -14,12 +14,20 @@ local T = require("ffi/util").template
 
 local Ftp = {}
 
+local function FTPurlEncode(str)
+    if not str then return nil end
+    return (str:gsub("([^%w%.%-_%~:/@])", function(c)
+        return string.format("%%%02X", string.byte(c))
+    end))
+end
 function Ftp:run(address, user, pass, path)
```

Comment:
Only the spaces, not the newline. ^_^ (Maybe it was the GH suggestion interface that did something weird?)
```suggestion
end

function Ftp:run(address, user, pass, path)
```